### PR TITLE
Refs [TASK][48858] Create splash screen. logo, theme

### PR DESCRIPTION
**Pull request by hungnd-3027**
## Related Tickets
- [#Task 48858: Create splash screen, logo, theme](https://edu-redmine.sun-asterisk.vn/issues/48858)
## WHAT
- Create splash screen, logo, theme, tabbar
## Evidence (Screenshot or Video)
Screenshot from 2022-05-06 17-29-22

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -6,20 +6,25 @@
...
         android:label="@string/app_name"
-        android:roundIcon="@mipmap/ic_launcher_round"
+        android:roundIcon="@drawable/ic_launch"
         android:supportsRtl="true"
         android:theme="@style/Theme.CooktailDB">
         <activity
-            android:name=".MainActivity"
-            android:exported="true">
+            android:name=".ui.SplashScreenActivity"
+            android:exported="true"
```
Comment (VietKFC) [inline app/src/main/AndroidManifest.xml:16]: bổ sung orientation portrait vào các màn hình nhé

File: app/src/main/java/com/example/cooktaildb/ui/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        super.onCreate(savedInstanceState)
+        binding?.bottomNavigationView?.setOnItemSelectedListener(this)
+
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.add(R.id.frame_navigation, HomeFragment.newInstance()).commit()
+    }
+
+    private fun replaceFragment(fragment: Fragment){
+        val transaction = supportFragmentManager.beginTransaction()
+        transaction.replace(R.id.frame_navigation, fragment).commit()
```
Comment (VietKFC) [inline app/src/main/java/com/example/cooktaildb/ui/MainActivity.kt]: viết ra extension để tái sử dụng nhé

Comment (VietKFC) [reply app/src/main/java/com/example/cooktaildb/ui/MainActivity.kt]: add to backstack ?

File: app/src/main/java/com/example/cooktaildb/ui/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            R.id.menu_filter -> {
+                replaceFragment(FilterFragment.newInstance())
+                return true
+            }
+            R.id.menu_favorite -> {
+                replaceFragment(FavoriteFragment.newInstance())
+                return true
+            }
+        }
+        return false
```
Comment (VietKFC) [inline app/src/main/java/com/example/cooktaildb/ui/MainActivity.kt]:
```suggestion
      return  when(item.itemId){
            R.id.menu_home -> {
                replaceFragment(HomeFragment.newInstance())
                true
            }
            R.id.menu_filter -> {
                replaceFragment(FilterFragment.newInstance())
                true
            }
            R.id.menu_favorite -> {
                replaceFragment(FavoriteFragment.newInstance())
                true
            }
            else -> false
        }
```

File: app/src/main/java/com/example/cooktaildb/ui/SplashScreenActivity.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.example.cooktaildb.ui
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+
+class SplashScreenActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
```
Comment (VietKFC) [inline app/src/main/java/com/example/cooktaildb/ui/SplashScreenActivity.kt]: remove blank

File: app/src/main/java/com/example/cooktaildb/ui/favorite_fragment/FavoriteFragment.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.example.cooktaildb.ui.favorite_fragment
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.cooktaildb.R
+
+class FavoriteFragment : Fragment() {
```
Comment (VietKFC) [inline app/src/main/java/com/example/cooktaildb/ui/favorite_fragment/FavoriteFragment.kt]: BaseFragment

File: app/src/main/java/com/example/cooktaildb/ui/filter_fragment/FilterFragment.kt [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package com.example.cooktaildb.ui.filter_fragment
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.cooktaildb.R
+
+class FilterFragment : Fragment() {
```
Comment (VietKFC) [inline app/src/main/java/com/example/cooktaildb/ui/filter_fragment/FilterFragment.kt]: tương tự

File: app/src/main/res/menu/activity_main.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    <item android:id="@+id/menu_filter"
+        android:icon="@drawable/ic_menu_filter"
+        android:title="@string/title_filter"/>
+
+    <item android:id="@+id/menu_favorite"
+        android:icon="@drawable/ic_menu_favorite"
+        android:title="@string/title_favorite"/>
+
+
+</menu>
```
Comment (VietKFC) [inline app/src/main/res/menu/activity_main.xml]: endline

File: app/src/main/res/values/themes.xml [hunk 8]
```diff
@@ -13,4 +13,8 @@
         <item name="android:statusBarColor" tools:targetApi="l">?attr/colorPrimaryVariant</item>
         <!-- Customize your theme here. -->
     </style>
+
+    <style name="SplashscreenTheme" parent="Theme.AppCompat.Light.NoActionBar">
```
Comment (VietKFC) [inline app/src/main/res/values/themes.xml]:
```suggestion
    <style name="SplashscreenTheme" parent="Theme.CocktailDB">
```

File: app/src/main/java/com/example/cooktaildb/ui/FragmentOperation.kt [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
+
+inline fun FragmentManager.inTransaction(func: FragmentTransaction.() -> Unit) {
+    val fragmentTransaction = beginTransaction()
+    fragmentTransaction.func()
+    fragmentTransaction.commit()
+}
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/cooktaildb/ui/FragmentOperation.kt]: Cái này ko cần thiết phải viết extension đâu ông , với lại extension của cái gì thì đặt tên class dạng AppcompatActivityExtension nhé

**Review by vietnh-1109 — APPROVED** [review]

**Pull request merged**